theano/gpuarray/opt.py

```diff
@@ -2104,6 +2119,17 @@ def local_gpu_svd(op, context_name, inputs, outputs):
         return op(inputs[0].astype('float32')).astype('float16')
     return op
 
+
+@register_opt('ctc', 'fast_compile')
+@op_lifter([theano.tensor.nnet.ctc.ConnectionistTemporalClassification])
+@register_opt2([ConnectionistTemporalClassification], 'ctc', 'fast_compile')
+def local_gpu_ctc(op, context_name, inputs, outputs):
+    if not config.ctc.enabled:
+        return
+    op = GpuConnectionistTemporalClassification(compute_grad=op.compute_grad)
+    return list(op(*inputs))
```

Comment:
As these ops set `default_output` to 0, only cost will be always returned when calling `op(*inputs)`, leading to only 1 output, but current node to be optimized may have 2 outputs (cost and grad) if compute_grad is true.

It's better to directly returned the result of `make_node()`, as this function will effectively return 2 outputs if compute_grad is true, and 1 output if compute_grad is false, so that we will always have exactly the same number of outputs as the current optimized node (otherwise optimization will fail).

```python
apply_node = op.make_node(*inputs)
return apply_node.outputs
```